### PR TITLE
[Feature1] Add object_test.go file

### PR DESCRIPTION
**Pull request by amiraminbeidokhti**
Added object_test.go file.
This is a Unit test file which has four functions.
coverage: 100.0% of statements

### CODE REVIEW

**Comment by shayan4shayan** [comment]
It's better to include wrong cases and check if system throws error when it should be throwing.

**Comment by amiraminbeidokhti** [comment]
> It's better to include wrong cases and check if system throws error when it should be throwing.

Thank you for your review.
There are several wrong cases for each function.
Would you please be more specific?

**Comment by shayan4shayan** [comment]
@amiraminbeidokhti 
first of all, I didn't understand what exactly are you testing. 
usually you have a module or code that you want to make sure it's working fine. so you writing test and using that code in your test, but you are not using any code from other parts of the project.

and also it's better to have more specific functions for testing,
for example 'SingleValidKey', 'CoupleValidKey', 'ListValidKey', 'SingleInvalidKey' , ...

**Comment by alidadar7676** [comment]
@shayan4shayan 
Hi, Thanks a lot for your tips. The purpose of these tests is to check the functionality of every single function. After this step, we start to write module-based tests

File: object/object_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		want string
+	}{
+		{Key{"Type", "Namespace", "Name"}, "{Type: Type, Namespace: Namespace, Name: Name}"},
+		{Key{"", "", ""}, "{Type: , Namespace: , Name: }"},
+		{Key{}, "{Type: , Namespace: , Name: }"},
+		{Key{Type: "t"}, "{Type: t, Namespace: , Name: }"},
+	}
+
+	for _, test := range tests {
+
```
Comment (alidadar7676) [inline object/object_test.go:20]: Remove extra white spaces

File: object/object_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		got := test.key1.Match(test.key2)
+
+		if got != test.want {
+			t.Errorf("got %v, want %v", got, test.want)
+		}
+	}
+}
+
+func TestObjectString(t *testing.T) {
```
Comment (alidadar7676) [inline object/object_test.go:85]: TestObjectStringer

File: object/object_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
+package object
+
+import (
+	"testing"
+)
+
+func TestKeyString(t *testing.T) {
```
Comment (alidadar7676) [inline object/object_test.go:7]: TestKeyStringer

**Pull request merged**